### PR TITLE
fix: M2-3039 Coordinator/Reviewer is able to 'Add to Builder' to the shared workspace

### PR DESCRIPTION
**Pull request by IhorNesterenko99**

### CODE REVIEW

File: src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.tsx [hunk 1]
```diff
@@ -32,21 +33,21 @@ export const AddToBuilderPopup = ({
...
   const [applets, setApplets] = useState<Applet[]>([]);
-  const { result: workspacesData = [] } = workspaces.useWorkspacesData() || {};
-  const currentWorkspace = workspaces.useData();
+  const workspacesWithRoles = workspacesState.useWorkspacesRolesData();
+  const workspaces =
+    workspacesWithRoles?.filter(
+      (workspace) =>
+        Object.keys(workspace.workspaceRoles).length === 0 ||
+        Object.values(workspace.workspaceRoles).some(
+          (roles) => roles.includes(Roles.Owner) || roles.includes(Roles.Manager),
```
Comment (kdanilenka32) [inline src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.tsx:45]: we have utils ownreormanger check

Comment (kdanilenka32) [reply src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.tsx:45]: and as I remember owner and manager always on 0 index

File: src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx [hunk 2]
```diff
@@ -265,8 +267,8 @@ export const getSteps = ({
 };
 
 export const getArrayFromApplets = (applets: FullApplet[]) =>
-  applets.reduce((acc: Applet[], { id, type, displayName, image }) => {
-    if (type === DashboardAppletType.Applet) {
+  applets.reduce((acc: Applet[], { id, type, displayName, image, role }) => {
+    if (type === DashboardAppletType.Applet && (role === Roles.Owner || role === Roles.Manager)) {
```
Comment (kdanilenka32) [inline src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx:271]: same here

Comment (IhorNesterenko99) [reply src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.utils.tsx:271]: as I understand in applets list there is only one main role which is a string
image

File: src/shared/components/Spinner/Spinner.styles.ts [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
+import { shouldForwardProp } from 'shared/utils';
 
-export const StyledSpinner = styled(StyledFlexAllCenter)`
+export const StyledSpinner = styled(StyledFlexAllCenter, shouldForwardProp)`
   position: absolute;
   width: 100%;
   height: 100%;
-  background-color: rgb(255 255 255 / 50%);
+  background-color: ${({ noBackground }: { noBackground?: boolean }) =>
+    noBackground ? 'transparent' : 'rgb(255 255 255 50%)'};
```
Comment (nmialik) [inline src/shared/components/Spinner/Spinner.styles.ts:11]: 'rgb(255 255 255 50%)' - in this format the color is displayed? no need for a slash 'rgb(255 255 255 / 50%)' ?
and maybe it makes sense to move it to variables?

**Review by kdanilenka32 — APPROVED** [review]

**Review by nmialik — APPROVED** [review]

**Pull request merged**